src/modules/keyboardmanager/common/KeyboardManagerState.cpp

```diff
@@ -166,12 +166,12 @@ void KeyboardManagerState::UpdateDetectShortcutUI()
     // Save the latest displayed shortcut
     currentShortcut = detectedShortcut;
     currentShortcut_lock.unlock();
-    std::vector<hstring> shortcut = detectedShortcut.GetKeyVector(keyboardMap);
-    
     detectedShortcut_lock.unlock();
 
     // Since this function is invoked from the back-end thread, in order to update the UI the dispatcher must be used.
-    currentShortcutUI.Dispatcher().RunAsync(Windows::UI::Core::CoreDispatcherPriority::Normal, [this, shortcut]() {
+    currentShortcutUI.Dispatcher().RunAsync(Windows::UI::Core::CoreDispatcherPriority::Normal, [this]() {
+
+        std::vector<hstring> shortcut = detectedShortcut.GetKeyVector(keyboardMap);

```

Comment:
Is there a reason for moving this here? Adding that on the UI thread could add more overhead, and we would also have to add a mutex lock and unlock here as well in that case.

Reply:
I see that it doesn't get updated if it's in the original line. Any idea why?

Reply:
Did some debugging, and to me it looks like GetKeyboardLayout gets the unchanged layout when called with arg 0 on the runner thread, I have no idea why though

Reply:
Yeah exactly, it seems that GetKeyboardLayout needs to be called from the UI thread for it to be updated. Probably related to the overall WM_INPUTLANGUAGECHANGED weird behaviour.

Reply:
Yeah. I don't think its a Xaml island issue though, it's probably an issue of how this is handled.

Reply:
I think the code can be kept like this for now then. If we are able to get the message method working later on then this will also be resolved.